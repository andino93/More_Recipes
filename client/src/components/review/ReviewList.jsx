import React from 'react';
import PropTypes from 'react-proptypes';
import moment from 'moment';

/**
 * Review
 * @param { object } props
 * @returns { object } object
 */
const ReviewList = props => (
  <div>
    <div className="row justify-content-center">
      <div className="card mb-10 col-lg-11 p-0">
        <div className="card-header">
          {props.review.users.username.toUpperCase()}
        </div>
        <div className="card-body">
          <blockquote className="blockquote mb-0">
            <p>
              <small>{props.review.content}</small>
            </p>
            <footer className="blockquote-footer">Posted &nbsp;
              <cite title="Source Title"> 
                {moment.utc(new Date(props.review.createdAt)).fromNow()}
              </cite>
            </footer>
          </blockquote>
        </div>
      </div>
    </div>
  </div>
);

ReviewList.propTypes = {
  review: PropTypes.shape().isRequired,
};

export default ReviewList;