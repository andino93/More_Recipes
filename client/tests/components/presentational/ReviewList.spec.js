import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import ReviewList from '../../../src/components/review/ReviewList.jsx';

configure({ adapter: new Adapter() });

describe('<ReviewList>', () => {
  const store = mockStore({});
  const connectedReviewList = mountWrap(ReviewList, store, { review: {
    users: {
      username: 'Justice'
    }
  } });
  it('should render the ReviewList Component', (done) => {
    expect(toJson(connectedReviewList)).toMatchSnapshot();
    done();
  });
});
