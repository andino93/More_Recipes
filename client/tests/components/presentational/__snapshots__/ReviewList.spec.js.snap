// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList> should render the ReviewList Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ReviewList
        review={
          Object {
            "users": Object {
              "username": "Justice",
            },
          }
        }
      >
        <div>
          <div
            className="row justify-content-center"
          >
            <div
              className="card mb-10 col-lg-11 p-0"
            >
              <div
                className="card-header"
              >
                JUSTICE
              </div>
              <div
                className="card-body"
              >
                <blockquote
                  className="blockquote mb-0"
                >
                  <p>
                    <small />
                  </p>
                  <footer
                    className="blockquote-footer"
                  >
                    Posted Â 
                    <cite
                      title="Source Title"
                    >
                      Invalid date
                    </cite>
                  </footer>
                </blockquote>
              </div>
            </div>
          </div>
        </div>
      </ReviewList>
    </Router>
  </BrowserRouter>
</Provider>
`;
